src/Raven.Client/Documents/Session/AsyncSessionDocumentTimeSeries.cs

```diff
@@ -173,11 +207,17 @@ await ServeFromCacheOrGetMissingPartsFromServerAndMerge(from ?? DateTime.MinValu
             {
                 if (ranges[toRangeIndex].From <= from)
                 {
-                    if (ranges[toRangeIndex].To >= to)
+                    if ((ranges[toRangeIndex].To >= to) || (ranges[toRangeIndex].Entries.Length > pageSize))
```

Comment:
Why do I care about the page-size here?

Reply:
If we have in cache range 1-5 with 3 entries and we ask for range 1-7 with page size 3 so we don't care that the range is bigger then what we have because we have all the results we need